core/templates/services/svg-sanitizer.service.ts

```diff
@@ -180,7 +180,7 @@ export class SvgSanitizerService {
     svg.querySelectorAll('*').forEach((node) => {
       let nodeTagName: string = node.tagName.toLowerCase();
       if (tagsToBeRemoved.indexOf(nodeTagName) !== -1) {
-        // The parent node will always be the svg tag itself.
+        // The outermost node of an SVG will always be the svg tag itself.
```

Comment:
Isn't there a way to assert that parentNode will always exist? Bypassing it this way seems to convey the wrong impression to developers -- it makes it look like node.parentNode is optional which just shouldn't be the case.

/cc @vojtechjelinek @mridul-netizen who might be able to offer more guidance.

Reply:
We could add a comment describing why the parent node will always exist and then use `@ts-ignore`.

Reply:
No, I'm pretty sure there's a way to tell typescript that something is not null. I suggest looking for it.

Reply:
Normally `!` operator asserts that something is not null. But ig Vojta could suggest a better alternative.

Reply:
Well, you don't seem to be using the `!` operator in the first place...

Reply:
That's because I would then have to bypass another eslint-check. :sweat:
![Screenshot from 2022-03-10 13-53-21](https://user-images.githubusercontent.com/89375125/157619443-57efe788-646e-465b-93fd-deb0ce52cc24.png)

Reply:
OK. Then let's ask @vojtechjelinek and see what he says.

Reply:
Can we add an extra `if` check here?

Reply:
Instead of using `parentNode!` or `parentNode?`, just do a normal if like `parentNode === undefined` (or `null`). If we know that `parentNode` will be set here and we don't expect `null` or `undefined` you can even throw an error if we find out that `parentNode` is `null` or `undefined`

Reply:
Right. Am on it!

Reply:
I don't think this is what @vojtechjelinek asked you to do. Please read his comment carefully and try again.

What you're doing here still has semantics of "node.parentNode may or may not exist", which is not the aim here.

Reply:
Hi @seanlip! Did some digging and found that `.remove()` method also exists on XML documents. I think it would be a better alternative as it avoids calling of the parent node and thus provides no case for null value to arise. Sorry for the inconvenience.